### ISSUE TITLE
Anonymous controller cannot define a custom route for custom controller

### ISSUE THREAD

**Issue by delner** [rspec/rspec-rails#1670]
**TL;DR** Renaming the controller under test from `ApplicationController` to `CustomController` breaks the example.

Given a fresh Rails 4.2.6 application, [the anonymous controller code sample provided by Relish for RSpec 3.5](https://www.relishapp.com/rspec/rspec-rails/docs/controller-specs/anonymous-controller):

```ruby
require "rails_helper"

RSpec.describe ApplicationController, :type => :controller do
  controller do
    def custom
      render :text => "custom called"
    end
  end

  specify "manually draw the route to request a custom action" do
    routes.draw { get "custom" => "anonymous#custom" }

    get :custom
    expect(response.body).to eq "custom called"
  end
end
```

...this passes. However, if you do the same example, but from a controller that is not `ApplicationController`, such as `CustomController` defined in `custom_controller.rb`:

```ruby
class CustomController < ActionController::Base
  # Prevent CSRF attacks by raising an exception.
  # For APIs, you may want to use :null_session instead.
  protect_from_forgery with: :exception
end
```

Then we'd expect the same example to pass:

```ruby
require "rails_helper"

RSpec.describe CustomController, :type => :controller do
  controller do
    def custom
      render :text => "custom called"
    end
  end

  specify "manually draw the route to request a custom action" do
    routes.draw { get "custom" => "anonymous#custom" }

    get :custom
    expect(response.body).to eq "custom called"
  end
end
```

But it fails with:

```
Failures:

  1) CustomController manually draw the route to request a custom action
     Failure/Error: get :custom
     
     ActionController::UrlGenerationError:
       No route matches {:action=>"custom", :controller=>"custom"}
     # ./spec/controllers/custom_controller_spec.rb:13:in `block (2 levels) in <top (required)>'
```

Am I doing something obviously wrong? Or is this broken?

**Comment by JonRowe** [comment]
Sorry it's not clear from the docs, but you need to draw your route using your custom name:

```Ruby
routes.draw { get "custom" => "custom#custom" }
```

**Comment by delner** [comment]
Thanks Jon. Making that adjustment makes the test pass. For the documentation, it might be helpful if there was another example on Relish that included this case with `routes.draw { get "custom" => "custom#custom" }`.

**Comment by delner** [comment]
It appears someone pointed this issue out previously in #866

**Comment by samphippen** [comment]
@delner thanks for opening the issue. Can you turn your sample code pasted above into a repository we can clone? That makes it much easier for us to debug. Typically, I ask people to follow these steps:

Thanks for the issue. We need a little more detail to be able to reproduce this.

Could you please provide us with a rails app that we can clone that demonstrates the issue. Specifically it'd be great if

1)  you could `rails new` an application and commit
2) make all the changes necessary to reproduce the issue and commit

then, provide us with a description of how to clone your application and reproduce the issue.

Thanks :)

**Comment by JonRowe** [comment]
@samphippen as above, I've already found that this is by design within the anonymous controller definition, the appropriate action here is to update the documentation for this caveat, no reproduction needed @delner

**Comment by samphippen** [comment]
Ah OK! My bad.
